File: legacy/node/src/utils/fileManager.js
const fs = require('fs');
const path = require('path');
const chalk = require('chalk');

// Chemins des fichiers
const CONFIG_PATH = path.join(__dirname, '../../config.json');
const HISTORY_PATH = path.join(__dirname, '../../data/history.json');
const ASSIGNMENTS_PATH = path.join(__dirname, '../../data/assignments.json');
const TOTEM_PATH = path.join(__dirname, '../../data/totem.json');

// Charger les données JSON
function loadJSON(filePath) {
  try {
    const data = fs.readFileSync(filePath, 'utf8');
    return JSON.parse(data);
  } catch (error) {
    console.error(chalk.red(`❌ Erreur lors de la lecture de ${filePath}:`), error.message);
    process.exit(1);
  }
}

// Sauvegarder les données JSON
function saveJSON(filePath, data) {
  try {
    fs.writeFileSync(filePath, JSON.stringify(data, null, 2), 'utf8');
  } catch (error) {
    console.error(chalk.red(`❌ Erreur lors de l'écriture de ${filePath}:`), error.message);
    process.exit(1);
  }
}

module.exports = {
  CONFIG_PATH,
  HISTORY_PATH,
  ASSIGNMENTS_PATH,
  TOTEM_PATH,
  loadJSON,
  saveJSON
};

File: chore_manager/utils/file_manager.py
import json
import sys
from pathlib import Path

from .console import get_console

ROOT_DIR = Path(__file__).resolve().parents[2]
CONFIG_PATH = ROOT_DIR / "config.json"
HISTORY_PATH = ROOT_DIR / "data" / "history.json"
ASSIGNMENTS_PATH = ROOT_DIR / "data" / "assignments.json"
TOTEM_PATH = ROOT_DIR / "data" / "totem.json"


def load_json(file_path: Path):
    console = get_console()
    try:
        data = file_path.read_text(encoding="utf-8")
        return json.loads(data)
    except Exception as exc:
        console.print(f"[red]❌ Erreur lors de la lecture de {file_path}:[/] {exc}")
        sys.exit(1)


def save_json(file_path: Path, data) -> None:
    console = get_console()
    try:
        file_path.write_text(json.dumps(data, indent=2, ensure_ascii=False), encoding="utf-8")
    except Exception as exc:
        console.print(f"[red]❌ Erreur lors de l'écriture de {file_path}:[/] {exc}")
        sys.exit(1)
